### PR TITLE
session,parser: make MAX_EXECUTION_TIME sql hint and global variable work

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/10955

### What is changed and how it works?

update go.mod to use the latest parser which fixes the SQL hint for MAX_EXECUTION_TIME
put `max_execution_time` into the auto loaded system variable list so it's correctly initialized

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by jackysp** [comment]
`go mod tidy` failed. PTAL @tiancaiamao

**Comment by tiancaiamao** [comment]
CI fixed. PTAL @jackysp @lysu

File: go.mod [hunk 1]
```diff
@@ -43,7 +43,7 @@ require (
 	github.com/pingcap/goleveldb v0.0.0-20171020122428-b9ff6c35079e
 	github.com/pingcap/kvproto v0.0.0-20190619024611-a4759dfe3753
 	github.com/pingcap/log v0.0.0-20190307075452-bd41d9273596
-	github.com/pingcap/parser v0.0.0-20190613082312-d2cf6071823d
+	github.com/pingcap/parser v0.0.0-20190627064259-e9c42442aa72
```
Comment (crazycs520) [inline go.mod]: Why need to change parser?

Comment (tiancaiamao) [reply go.mod]: To include this fix https://github.com/pingcap/parser/pull/366
The SQL hint is not successful parsed in `select /*+ MAX_EXECUTION_TIME(1000) */ SLEEP(5);`
@crazycs520

**Pull request merged**